features/cellular/framework/device/CellularStateMachine.cpp

```diff
@@ -329,8 +292,10 @@ void CellularStateMachine::retry_state_or_fail()
 
 void CellularStateMachine::state_init()
 {
-    _cellularDevice.set_timeout(TIMEOUT_POWER_ON);
-    tr_info("Start connecting (timeout %d s)", TIMEOUT_POWER_ON / 1000);
+#if MBED_CONF_CELLULAR_TIMEOUT_POWER_ON
```

Comment:
Why this MACRO everywhere? Why not define set_timeout as empty function?

Reply:
This is to disable set_timeout on `CellularStateMachine`, while it should be possible for application to use `set_timeout` via `CellularDevice` API.

Reply:
As we have a default value, the user can override that with NULL to "disable" it. What would go wrong if you call set_timeout with NULL and in function body does nothing in that case? User can call that with NULL as well, so we should have NULL check already in place, or?